addons/sale_stock/models/stock.py

```diff
@@ -126,6 +126,8 @@ def _prepare_procurement_values(self):
         # to pass sale_line_id fom SO to MO in mto
         if self.sale_line_id:
             res['sale_line_id'] = self.sale_line_id.id
+            if self.sale_line_id.route_ids:
+                res['route_ids'] += self.sale_line_id.route_ids
```

Comment:
```suggestion
                res['route_ids'] |= self.sale_line_id.route_ids
```
Let's say you have:
- res['route_ids'] = `stock.route(1, 2)`
- self.sale_line_ids.route_ids = `stock.route(2, 3)`

Then:
- `res['route_ids'] += self.sale_line_id.route_ids` -> `stock.route(1, 2, 2, 3)`
- `res['route_ids'] |= self.sale_line_id.route_ids` -> `stock.route(1, 2, 3)`